### PR TITLE
replace xhr with d2lfetch

### PR DESCRIPTION
**Pull request by CodeBaboon**
This should probably be a thing to get better perf. I know Adam expressed concerns around potential Safari issues so we should test it pretty thoroughly on the Safari's before we merge it in.

### CODE REVIEW

**Comment by ryantmer** [comment]
I can take a look if nobody else is - one of the OSX VMs has just been updated, so it's got the latest and greatest Safari on it :)

File: bower.json [hunk 1]
```diff
@@ -13,7 +13,8 @@
     "d2l-simple-overlay": "Brightspace/simple-overlay#1.1.0",
     "d2l-typography": "^5.3.0",
     "iron-icon": "^1.0.12",
-    "polymer": "Polymer/polymer#^1.9.2"
+    "polymer": "Polymer/polymer#^1.9.2",
+    "fetch": "^2.0.3"
```
Comment (ryantmer) [inline bower.json]: The nittiest nit: other items here are alphabetical

(so very nit)

File: behaviors/d2l-upcoming-assessments-behavior.html [hunk 2]
```diff
@@ -2,6 +2,8 @@
 <link rel="import" href="../../d2l-hm-constants-behavior/d2l-hm-constants-behavior.html">
 <link rel="import" href="./date-behavior.html">
 
+<script src="../bower_components/fetch/fetch.js"></script>
+<script src="https://s.brightspace.com/lib/d2lfetch/1.0.0/d2lfetch.js"></script>
```
Comment (ryantmer) [inline behaviors/d2l-upcoming-assessments-behavior.html]: Doesn't look like it, but is there anything we want in the newer versions? (Although I guess that would mean everything ends up using the newer version, ja?)

Comment (CodeBaboon) [inline behaviors/d2l-upcoming-assessments-behavior.html]: Yeah we will want 1.0.2, good point. Didn't exist when I committed this originally :)

**Comment by ryantmer** [comment]
For information's sake: tested this out, spoke with @CodeBaboon and it seems the crux of the issue is [this little gem here](https://developer.mozilla.org/en-US/docs/Web/API/Response/clone#Browser_compatibility) - the dedupe middleware uses `clone`, but Safari doesn't support it. The polyfill backs off because `window.fetch` is already a thing, though, so that means it doesn't polyfill just `.clone` of course.

Yaaaay.

Possible course of action is to forego the use of `clone` and just copy the Response manually in the dedupe middleware.

**Comment by ChrisKraljevicD2L** [comment]
Code Review Checklist
- [ ] Verify that unit tests and spec tests have been written, if appropriate
- [ ] Verify that CI checks are passing
- [ ] Verify that the code is behind a feature flag
- [ ] Verify that appropriate documentation has been done (readme, inline, docs.dev, etc.) and makes sense
- [ ] Verify to the best of your ability that no security/privacy flaws are present
- [ ] Verify that accessibility standards are held to
- [ ] Verify that no lang terms are hard coded
- [ ] Verify that lang terms have been submitted for translation
- [ ] Other review for coding style, functionality, and best practices

Testing Checklist
Sanity Checks
- [ ] Verify that unit tests and spec tests have been written, if appropriate
- [ ] Verify that unit tests, spec tests, and linting are all passing
- [ ] Verify that no code review comments are outstanding
- [ ] Verify that the code is behind a feature flag

Acceptance/Functionality
- [ ] Check that the acceptance criteria of the story has been fulfilled
- [ ] Check that related functionality has not been severely affected (smoke test)
- [ ] (UI) Check that the functionality is the same across browsers and devices

Accessibility (UI)
- [ ] Verify that workflows can be completed via screen reader/keyboard
- [ ] Verify that workflows can be completed when the browser is zoomed to 200%
- [ ] Verify that there are no obvious colour contrast/combination problems

Internationalization (UI)
- [ ] Verify that lang terms are not hard coded
- [ ] Verify that workflows can be completed when RTL

**Comment by ChrisKraljevicD2L** [comment]
Testing notes:

- Safari on iPad works now, functionality not affected in other browsers, exceeeept...

- Edge is now borked. "Oops! There was an issue loading this page. Don't worry, we're on it." When loading the page, reloading after that results in a blank page. No errors in console (that I could see)

**Pull request merged**